### ISSUE TITLE
How can I Evaluate images using TransferLearning.model in a C# project.

### ISSUE THREAD

**Issue by yuanweilei** [Microsoft/CNTK#1676]
When I use
 evaluatorFunc.Evaluate(inputDataMap, outputDataMap, DeviceDescriptor.CPUDevice);
to evaluate images, it failed and error message as follow:
    The only permitted binary reduction operation is opSum

I trained a TransferLearning.model by CNTK-GPU v2.0-b eta15 and want to use it in a C# project.
And I use CNTK.CPUOnly nuget.
Do  I should use CNTK.GPU nuget?

**Comment by zhouwangzw** [comment]
@yuanweilei  How did you train your model, using python? You can try with CNTK.GPU nuget, but probably will get the same error. Would it be possible to share a small repro?

Thanks,

**Comment by yuanweilei** [comment]
@zhouwangzw I trained the model using Python (TransferLearning.py) on a Azure VM (which has a GPU device). And then I copy the model to my computer (which has not GPU). If possible, I want to use the model in my computer.
I will try with CNTK.GPU, thank you.

By the way, what does "a small repro" mean? (I am very sorry, I don not know "repro" )

**Comment by pkranen** [comment]
If you used ResNet as the base model (as in the Transfer Learning example and tutorial) then you need a GPU since ResNet contains batch normalization layers which are not supported on CPU currently.

**Comment by yuanweilei** [comment]
@zhouwangzw Could you tell me CNTK.CPU/CNTK.GPU nuget support TransferLearning.model? 
Does there are sample codes of TransferLearning.model using C#?

**Comment by yuanweilei** [comment]
@pkranen Thank you very much! I used ResNet model. 
I will try to use a different base model.
Can you tell me which model I can use if I want to use CPU nuget please?

**Comment by pkranen** [comment]
You can use [AlexNet](https://github.com/Microsoft/CNTK/tree/master/Examples/Image/Classification/AlexNet#pre-trained-models) or [VGG](https://github.com/Microsoft/CNTK/tree/master/Examples/Image/Classification/VGG#pre-trained-models). How to use a different base model is described [here](https://github.com/Microsoft/CNTK/wiki/Build-your-own-image-classifier-using-Transfer-Learning#using-a-different-base-model).

**Comment by zhouwangzw** [comment]
The latest CNTK vesion should support BatchNormalization on CPU (#1005).
We have [C# samples ](https://github.com/Microsoft/CNTK/blob/master/Examples/Evaluation/CNTKLibraryCSEvalCPUOnlyExamples/CNTKLibraryCSEvalExamples.cs) for evaluating a pre-trained model, e.g. ResNet. Do you also want to evaluate an intermediate node with your transfering model?

**Comment by yuanweilei** [comment]
@zhouwangzw Thank you.
I only want to do like [EvaluationSingleImage](https://github.com/Microsoft/CNTK/blob/master/Examples/Evaluation/CNTKLibraryCSEvalCPUOnlyExamples/CNTKLibraryCSEvalExamples.cs)

I use TransferLearning model and only change this row,
`string modelFilePath = "TransferLearning.model";`

everything is ok but it failed on
`modelFunc.Evaluate(inputDataMap, outputDataMap,  DeviceDescriptor.CPUDevice);`

I will try to use CNTK-GPU nuget and use the same code.

**Comment by yuanweilei** [comment]
@pkranen Thank you very much!
I use CNTK-GPU nuget and I can evaluate images now.

Now my plan is: 
1. rent a Azure windows server (which has GPU like NC6), and create a WebAPI project to evaluate images using CNTK-GPU
2. create a project in my compute (which has not GPU) and use the WebAPI project.

I want to create some bots (using MS bot framework) to use CNTK or MS cognitive service finally.
I close this issues. Thank you.

**Comment by wendy7707** [comment]
@yuanweilei Hello, I have the same problem with you, so you means I must need GPU nuget? it is not supported CPU?  Thanks!
